M3 Trip Create/Edit

kanatagu opened the pull request:
# Description
 
* UI for create/edit
* API connecting for create/edit
* Fixed Date issue

**NOTE**
If you want to check in local environment, please change the backend branch to https://github.com/tabi-memo/backend/pull/18 first, then reset and seed.
```
cd Change_schema_date
pnpm supabase db reset
pnpm seed
```

Fixes # (issue)
https://github.com/tabi-memo/frontend/issues/20

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Check List

## URL
Create page: http://localhost:3000/trip/create
Edit Page (ex) : http://localhost:3000/trip/50370a01-879a-4e46-bda7-70e2022241cb/edit

## Screen Recording 
Create

https://github.com/tabi-memo/frontend/assets/66394413/8f4d7d68-5e7d-46b7-90f0-46c47d48f515

Edit
https://github.com/tabi-memo/frontend/assets/66394413/b8ef5549-2a84-4dc4-b487-6bb597b111d8

kanatagu committed:
Add Trip edit/create UI

kanatagu committed:
Merge branch 'main' into M3_trip_create

kanatagu committed:
Fix name

kanatagu committed:
Amend storybook

samuraikun reviewed (commented):
@Kana-TAGUCHI Date form doesn't seem to be set value :eyes:

![6d4260f312e1f6137f70de46c400458f](https://github.com/tabi-memo/frontend/assets/7115171/c9dfbbb0-bd41-4395-b7ab-96aa864b34d6)

samuraikun commented:
@Kana-TAGUCHI Let tell me the reason why doesn't image form upload and preview a file in this PR?

![4583ec59b16dc55a50a3dfb8cae15cf3](https://github.com/tabi-memo/frontend/assets/7115171/b6b87ab4-3423-4fb2-abd6-243952136f44)

kanatagu commented:
@samuraikun 
Yes because this is just layout PR! I implemented only layout and will work on setValue, Image upload and etc...
I implemented tags little otherwise you can not see the tags below the input.

YoshinariYamanaka commented on app/components/input/index.ts:
```diff
@@ -1,3 +1,3 @@
-export { InputSearch } from './input-search'
 export { InputForm } from './input-form'
+export { InputIcon } from './input-icon'
```
Is it ok to use relative path with dot ?
ref: https://github.com/tabi-memo/frontend/pull/7#discussion_r1406898337

But for me, it's ok to use especially when we import a file from the same directory.

kanatagu replied:
Yes... I realized that it is okay (or even better) to use relative path if it's on the same directory so I actually changed since I realized it. Sorry 🙏

YoshinariYamanaka commented on app/trip/[id]/edit/page.tsx:
```diff
@@ -0,0 +1,34 @@
+'use client'
+
+import { Box, Container, Heading, useColorModeValue } from '@chakra-ui/react'
+// import { Loading } from '@/components/loading'
```
I'd say it's better to delete comment out source, cuz it's unnecessary.

kanatagu replied:
I will use it next logic.. Can I just leave it? 🙏

YoshinariYamanaka commented on app/trip/components/trip-form.tsx:
```diff
@@ -0,0 +1,67 @@
+        </FormControl>
+
+        <FormControl>
+          <FormLabel>Date From</FormLabel>
+          <CustomDateTimePicker
+            onChange={() => console.log('onChange')}
+            value={null}
+          />
+        </FormControl>
+
+        <FormControl>
+          <FormLabel>Date To</FormLabel>
+          <CustomDateTimePicker
+            onChange={() => console.log('onChange')}
```
We might be able to clarify where `onChange` log comes from with stack trace on a browser, but I'm just wondering if it's worth logging out if you really need this log.

kanatagu replied:
This is just temporal... I will change it for the next logic PR

YoshinariYamanaka reviewed (commented):
I left just few comments !

SachiGoto commented on app/trip/components/trip-form.tsx:
```diff
@@ -0,0 +1,67 @@
+
+        <FormControl>
+          <FormLabel>Date From</FormLabel>
+          <CustomDateTimePicker
```
The date inputs on the trip page and activity page are slightly different. The date field on the trip page doesn't have a time option, whereas the activity page includes it. So, you may need to modify the custom date-time picker for the tip page.

The custom date-time picker has a calendar icon but based on the figma design, it should be a clock icon for the activity page , so I need to correct that. This means that the customDateTimePicker is not as reusable as intended and may need to move it under the activity page. 

What are your thoughts on this?

SachiGoto commented on app/trip/[id]/page.tsx:
```diff
@@ -31,7 +31,7 @@ export default function TripDetailsPage({
     <>
       <Header />
-      <Box as="main" minH="100vh" bg={bg} color={color}>
+      <Box as="main" minH="100svh" bg={bg} color={color}>
```
Interesting that you switch  from vh to svh! I've never used svh before. I'm just curious why svh would be better in this case?

SachiGoto commented on app/trip/create/page.tsx:
```diff
@@ -1,20 +1,27 @@
       <Box as="main" minH="100vh" bg={bg} color={color}>
         <Container
-          maxW={{ base: '100%', lg: 'container.xl' }}
+          maxW={{ base: '100%', lg: 'container.sm' }}
```
When the screen size is tablet size, it still takes up 100% of the width, and the input fields appear too wide. Perhaps consider adding container.sm starting from the medium size, like the example below:
`maxW={{ base: '100%', md: 'container.sm' }}`
<div style="display: flex; justify-content: flex-start; align-items: flex-start;">
  <img width="300" alt="Screenshot 1" src="https://github.com/tabi-memo/frontend/assets/70562492/b333270a-825d-478b-b77d-12f51d17de44">
  <img width="300" alt="Screenshot 2" src="https://github.com/tabi-memo/frontend/assets/70562492/634652eb-5840-4164-9b63-7da63d344bd8">
</div>

kanatagu committed:
Change tags form

kanatagu committed:
Change checkbox

kanatagu committed:
Add data logic related tags

kanatagu committed:
Create trip update mutation and custom hooks

kanatagu committed:
Add create trip form

kanatagu committed:
Run lint

kanatagu committed:
Amend dates display

kanatagu committed:
Display total trip cost

kanatagu committed:
Resolved conflict

kanatagu commented on app/components/date/custom-date-picker.tsx:
```diff
@@ -0,0 +1,106 @@
+import DatePicker from 'react-datepicker'
```
You see a warning but this is depending on DatePicker library...
<img width="635" alt="Screenshot 2024-01-30 at 12 14 26 PM" src="https://github.com/tabi-memo/frontend/assets/66394413/175616a1-9e47-47c9-9af2-e7617e61d322">

kanatagu commented on app/libs/utils/date.ts:
```diff
@@ -1,5 +1,42 @@
+import { toDate, utcToZonedTime, format } from 'date-fns-tz'
+
+/*
+ * NOTE: This timezone is used for the entire application.
+ * All form inputs for date or date time should have offset of +09:00 e.g."2024-01-01T00:00:00+09:00", which is Tokyo timezone.
+ * See "formatToISODate" function.
+ */
+const TIMEZONE = 'Asia/Tokyo'
```
This is important for this application so I added NOTE

kanatagu commented on app/trip/components/trip-form.tsx:
```diff
@@ -0,0 +1,236 @@
+          <FormErrorMessage>{errors?.date_to?.message}</FormErrorMessage>
+        </FormControl>
+
+        {/* TODO Image Upload to storage & Send the URL string to DB */}
```
Image will be implemented later after Koji's PR is merged!

kanatagu committed:
Add handling refetch trips collection

samuraikun reviewed (commented):
"Total Cost" section is slightly covered by the header. 👀 
![image](https://github.com/tabi-memo/frontend/assets/7115171/7510b25f-2d6f-478d-b061-7fa1ede395b7)

samuraikun reviewed (approved):
I only commented on one, but it looks good to me 👍 🍎

kanatagu committed:
Fix cost UI style

kanatagu commented:
@samuraikun 
Oh no thank you..! Sorry, I was too rushing and didn't check the smart phone in the end. 
I fixed it
https://github.com/tabi-memo/frontend/pull/25/commits/491d7d0df308e441d3c8dad7963a8137877e6ad4

<img width="474" alt="Screenshot 2024-01-31 at 8 39 14 PM" src="https://github.com/tabi-memo/frontend/assets/66394413/0e22f749-d9af-4b61-b6e0-28bb36add9ce">

<img width="466" alt="Screenshot 2024-01-31 at 8 39 21 PM" src="https://github.com/tabi-memo/frontend/assets/66394413/a13b0bb0-c60f-4419-a8f4-71086d9d997e">
